src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -1,50 +1,108 @@
+import {remoteData} from "shared/api/remoteData";
 
 export interface ICoExpressionTabProps {
     store:ResultsViewPageStore;
 }
 
-export class CoExpressionCache extends MobxPromiseCache<{entrezGeneId:number, molecularProfile:MolecularProfile, allData:boolean}, CoExpression[]> {};
+export class CoExpressionCache extends MobxPromiseCache<{profileX: MolecularProfile, 
+    profileY: MolecularProfile, geneticEntityId: string,
+    geneticEntityType: GeneticEntityType, allData:boolean}, CoExpression[]> {}
 
 @observer
 export default class CoExpressionTab extends React.Component<ICoExpressionTabProps, {}> {
-    @observable _selectedMolecularProfile:MolecularProfile|undefined; // only undefined initially, until molecular profiles downloaded
-    @observable _selectedEntrezGeneId:number | undefined; // only undefined initially, until genes downloaded
+    @observable _selectedProfileX:string|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedProfileY:string|undefined; // only undefined initially, until molecular profiles downloaded
+    @observable _selectedGeneticEntityId:string|undefined; // only undefined initially, until genes and gene sets downloaded
 
-    @computed get selectedMolecularProfile():MolecularProfile|undefined {
-        if (!this._selectedMolecularProfile && this.props.store.coexpressionTabMolecularProfiles.isComplete
-            && this.props.store.coexpressionTabMolecularProfiles.result.length > 0) {
-            return this.props.store.coexpressionTabMolecularProfiles.result[0];
-        } else {
-            return this._selectedMolecularProfile;
+    readonly selectedProfileX = remoteData<string|undefined>({
+        await: ()=>[
+            this.xProfileOptions
+        ],
+        invoke: () => {
+            if (!this._selectedProfileX && this.xProfileOptions.result!.length > 0) {
+                return Promise.resolve(this.xProfileOptions.result![0].value);
+            } else {
+                return Promise.resolve(this._selectedProfileX);
+            }
         }
-    }
+    });
 
-    @computed get selectedEntrezGeneId():number | undefined {
-        if (this._selectedEntrezGeneId === undefined && this.props.store.genes.isComplete &&
-                this.props.store.genes.result.length > 0) {
-            return this.props.store.genes.result[0].entrezGeneId;
-        } else {
-            return this._selectedEntrezGeneId;
+    readonly selectedProfileY = remoteData<string|undefined>({
+        await: ()=>[
+            this.yProfileOptions
+        ],
+        invoke: () => {
+            if (!this._selectedProfileY && this.yProfileOptions.result!.length > 0) {
+                return Promise.resolve(this.yProfileOptions.result![0].value);
+            } else {
+                return Promise.resolve(this._selectedProfileY);
+            }
         }
-    }
+    });
+
+    readonly selectedGeneticEntityId = remoteData<string>({
```

Comment:
why not just have a promise that returns the selectedEntity and consumers can take whatever property they need of that object?   Is it true that the _selectedGeneticEntityId will always be represented in the geneticEntities collection?